scylla/src/transport/session.rs

```diff
@@ -1592,12 +1775,21 @@ impl Session {
                             error = %e,
                             "Choosing connection failed"
                         );
+                        span.record(
+                            scylladb_tag!("completion_status"),
```

Comment:
I don't really like that the tag names are repeated and scattered over the codebase. It's easy to make a typo.

If you have some fields that can't be set immediately when the span is being created, maybe you could wrap the Span with a struct? That struct would provide methods that set appropriate fields. This would be much less prone to typo errors and would make it more clear which tags/values a span for a `query`/`execute`/`batch` can contain - all the tag names would be in one place, in the `impl` block.